addon/components/tf-button/component.js

```diff
@@ -0,0 +1,35 @@
+/**
+  @module ticketfly-buttons
+ */
+import Ember from 'ember';
+import layout from './template';
+import { collect } from 'ember-computed';
+const { Component, computed, get } = Ember;
+
+/**
+  @public
+  @class TfButton
+  @extends Ember.Component
+ */
+export default Component.extend({
+  layout,
+
+  tagName: 'button',
+  classNames: ['c-button'],
+  attributeBindings: ['disabled'],
+  classNameBindings: ['buttonCategory'],
+
+  utilityGroups: collect('weightGroup', 'colorGroup'),
+
+  buttonCategory: computed('utilityGroups', {
+    get() {
+      let buttonCategory = '';
+      get(this, 'utilityGroups').map((group) => {
+        if (group) {
+          buttonCategory += `u-button-${group} `;
```

Comment:
Actually, the **Block** in BEM shouldn't be the module name (ticketfly-buttons).

Some questions for clarity:

`tf-button` doesn't have any child elements, so it can simply be:
```html
<button class="tf-button tf-button--primary tf-button--positive">Click Me</button>
```

However, for `tf-icon-button`. Is the **Block** `tf-icon-button` or `tf-button`? 
```html
<button class="tf-icon-button tf-icon-button--secondary tf-icon-button--neutral">
  <span class="tf-icon-button__icon-plus"></span>
  <span>Add a thing</span>
</button>

<button class="tf-button tf-button--secondary tf-button--neutral">
  <span class="tf-button__icon-plus"></span>
  <span>Add a thing</span>
</button>
```

Some decision on the icon part would be good too. The options I see are:
```html
<span class="tf-button__icon-plus"></span>

<span class="tf-icon-button__icon--plus"></span>

<span class="tf-button__icon--plus"></span>
```

Perhaps this is clear to someone else. In my experience, everyone interprets BEM differently. If there's an internal doc or reference I can look at that'd be good too.